.gitlab-ci.yml

```diff
@@ -2339,34 +2384,47 @@ dca_dev_master_docker_hub:
   needs: ["fail_on_non_triggered_tag", "build_cluster_agent_amd64"]
   only:
     - master
-  variables:
-    <<: *docker_hub_variables
   script:
     - inv -e docker.publish --signed-push ${SRC_DCA}:${SRC_TAG}-amd64 datadog/cluster-agent-dev:master
 
 # deploys nightlies to agent-dev
-dev_nightly_docker_hub:
+dev_nightly_docker_hub-a6:
+  <<: *skip_when_unwanted_on_6
   <<: *docker_tag_job_definition
   <<: *run_when_triggered_on_nightly
   needs:
     - fail_on_non_triggered_tag
     - build_agent6
     - build_agent6_jmx
     - build_agent6_py2py3_jmx
-    - build_agent7
-    - build_agent7_jmx
-    - build_dogstatsd_amd64
-  variables:
-    <<: *docker_hub_variables
   script:
     - inv -e docker.publish --signed-push ${SRC_AGENT}:${SRC_TAG}-6-amd64       datadog/agent-dev:nightly-${CI_COMMIT_SHORT_SHA}
     - inv -e docker.publish --signed-push ${SRC_AGENT}:${SRC_TAG}-6-amd64       datadog/agent-dev:nightly-${CI_COMMIT_SHORT_SHA}-py2
-    - inv -e docker.publish --signed-push ${SRC_AGENT}:${SRC_TAG}-7-amd64       datadog/agent-dev:nightly-${CI_COMMIT_SHORT_SHA}-py3
     - inv -e docker.publish --signed-push ${SRC_AGENT}:${SRC_TAG}-6-jmx-amd64   datadog/agent-dev:nightly-${CI_COMMIT_SHORT_SHA}-jmx
     - inv -e docker.publish --signed-push ${SRC_AGENT}:${SRC_TAG}-6-jmx-amd64   datadog/agent-dev:nightly-${CI_COMMIT_SHORT_SHA}-py2-jmx
+
+# deploys nightlies to agent-dev
+dev_nightly_docker_hub-a7:
+  <<: *skip_when_unwanted_on_7
+  <<: *docker_tag_job_definition
+  <<: *run_when_triggered_on_nightly
+  needs:
+    - fail_on_non_triggered_tag
+    - build_agent7
+    - build_agent7_jmx
+  script:
+    - inv -e docker.publish --signed-push ${SRC_AGENT}:${SRC_TAG}-7-amd64       datadog/agent-dev:nightly-${CI_COMMIT_SHORT_SHA}-py3
     - inv -e docker.publish --signed-push ${SRC_AGENT}:${SRC_TAG}-7-jmx-amd64   datadog/agent-dev:nightly-${CI_COMMIT_SHORT_SHA}-py3-jmx
-    - inv -e docker.publish --signed-push ${SRC_DSD}:${SRC_TAG}-amd64           datadog/dogstatsd-dev:nightly-${CI_COMMIT_SHORT_SHA}
 
+# deploys nightlies to agent-dev
+dev_nightly_docker_hub-dogstatsd:
```

Comment:
Should this only be done on A7 pipelines? The prod dogstatsd docker deploy is only done on A7 pipelines, so it would make sense to do the same for other dogstatsd docker deploys (nightly, master, branch). That way it would also be in sync with the .deb package, which is only built on A7 pipelines.

Reply:
Actually we could merge dogstatsd and A7 now, but I think we might want to push just one or another image at some point.